libs/sdk-common/src/input_parser.rs

```diff
@@ -197,6 +226,69 @@ pub async fn parse(
     Err(anyhow!("Unrecognized input type"))
 }
 
+async fn bip353_parse(
+    input: &str,
+    dns_resolver: &AsyncResolver<GenericConnector<TokioRuntimeProvider>>,
+) -> Option<String> {
+    if let Some((local_part, domain)) = input.split_once('@') {
+        let dns_name = format!("{}.{}.{}", local_part, USER_BITCOIN_PAYMENT_PREFIX, domain);
+
+        // Query for TXT records of a domain
+        let txt_data = match dns_resolver.txt_lookup(dns_name).await {
```

Comment:
From the spec:

`Clients resolving Bitcoin payment instructions MUST ignore any TXT records at the same label which do not begin with (ignoring case) "bitcoin:". Resolvers encountering multiple "bitcoin:"-matching TXT records at the same label MUST treat the records as invalid and refuse to use any payment instructions therein.`

We need to make sure we validate that.